src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -107,15 +107,19 @@ const UsersFragment = ({
   }, []);
 
   React.useEffect(() => {
+    if (searchText?.length === 0) {
+
     handleActiveSections();
+    }
   }, [searchText, followedUsers]);
+  
 
   const handleActiveSections = () => {
     if (!searchText) return;
 
     if (searchText.length === 0) {
       setActiveSections([]);
-    } else if (searchText.length >= 0 && followedUsers.length > 0) {
+    } else if (searchText?.length >= 0 && followedUsers?.length > 0) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a logical inconsistency in the conditions here. The first condition checks if `searchText` is falsy, and if so, it returns immediately. The second condition checks if `searchText.length` is 0, which would only be true if `searchText` is an empty string. However, an empty string is a falsy value, so the first condition would have already returned in this case. Therefore, the second condition is unreachable and can be removed.
```diff
-     if (searchText.length === 0) {
-       setActiveSections([]);
-     } else 
      if (searchText?.length >= 0 && followedUsers?.length > 0) {
        setActiveSections([0]);
      } else {
        setActiveSections([]);
      }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->